ruby/lib/asciidoctor/extensions/asciidoctor_kroki/extension.rb

```diff
@@ -291,19 +291,22 @@ def save(output_dir_path, kroki_client)
       diagram_name = "diag-#{Digest::SHA256.hexdigest diagram_url}.#{@format}"
       file_path = File.join(output_dir_path, diagram_name)
       encoding = case @format
-                 when 'txt', 'atxt', 'utxt'
+                 when 'txt', 'atxt', 'utxt', 'svg'
                    'utf8'
-                 when 'svg', 'binary'
+                 else
                    'binary'
                  end
       # file is either (already) on the file system or we should read it from Kroki
-      contents = File.exist?(file_path) ? File.open(file_path, &:read) : kroki_client.get_image(self, encoding)
-      FileUtils.mkdir_p(output_dir_path)
-      if encoding == 'binary'
-        File.binwrite(file_path, contents)
-      else
-        File.write(file_path, contents)
+      unless File.exist?(file_path)
+        contents = kroki_client.get_image(self, encoding)
+        FileUtils.mkdir_p(output_dir_path)
+        if encoding == 'binary'
```

Comment:
Technically, you don't need this distinction for writing. As long as you are just passing through the content, you can use:

```rb
File.write file_path, contents, mode: 'wb'
```

That gets the job done just as efficiently in my experience.

In Ruby, the "binary" write mode is really more of a raw mode (kind of like it is in Node.js).